### PR TITLE
AKS: support apiserver-vnet-integration for public cluster

### PR DESCRIPTION
**Pull request by levimm**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Review by FumingZhang — COMMENTED** [review]
Queued a [pipeline](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=59900984&view=results) to test the newly added test case. The test failed due to a live incident, which is out of the scope of AKS.

Also 0.5.96 has just been merged, need to resolve some conflicts.

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 1]
```diff
@@ -1259,6 +1255,304 @@ def get_apiserver_subnet_id(self) -> Union[str, None]:
...
+        # read the original value passed by the command
+        enable_apiserver_vnet_integration = self.raw_param.get("enable_apiserver_vnet_integration")
+        disable_private_cluster = self.raw_param.get("disable_private_cluster")
+
+        # this parameter does not need dynamic completion
+        # validation
+        if enable_validation:
+            if self.decorator_mode == DecoratorMode.UPDATE:
+                # logic copied from cli core
+                if disable_private_cluster:
```
[the remark this answers is not in the document]
Comment (levimm) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:1364]: it's covered in `_get_enable_private_cluster` when it's a public cluster and tries to enable/disable public fqdn.

File: src/aks-preview/azext_aks_preview/_help.py [hunk 2]
```diff
@@ -811,6 +811,15 @@
...
           short-summary: Disable node restriction option on cluster.
+        - name: --enable-private-cluster
+          type: bool
+          short-summary: Enable private cluster for apiserver vnet integration cluster.
+        - name: --disable-private-cluster
+          type: bool
+          short-summary: Disable private cluster for apiserver vnet integration cluster.
+        - name: --private-dns-zone
+          type: string
+          short-summary: The private dns zone mode for private cluster.
```
[the remark this answers is not in the document]
Comment (levimm) [reply src/aks-preview/azext_aks_preview/_help.py:837]: added in param.

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (FumingZhang) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:1444]: Got it!

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**